### PR TITLE
Cog refactor

### PR DESCRIPTION
**Pull request by Just-Jojo**
## Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New Feature
- [ ] Other

## Description of the changes
Various touch-ups and some refactoring :D

### CODE REVIEW

File: advancedblacklist/advanced_blacklist.py [hunk 1]
```diff
@@ -94,25 +89,32 @@ async def localblacklist(self, ctx: commands.Context):
...
+        *,
+        reason: str = "No reason provided.",
     ):
-        """Add a user to this guild's blacklist"""
-        if user.bot:
-            return await ctx.send("You cannot add a bot to the blocklist.")
-        elif await self.bot.is_owner(user):
-            return await ctx.send("You cannot add an owner of this bot to the blocklist.")
-        elif ctx.guild.owner == user:
+        r"""Add a user to this guild's blacklist
```
Comment (Kreusada) [inline advancedblacklist/advanced_blacklist.py:98]:
```suggestion
        """Add a user to this guild's blacklist
```

File: advancedblacklist/advanced_blacklist.py [hunk 2]
```diff
@@ -266,15 +296,18 @@ async def blacklist_name_remove(self, ctx: commands.Context, *, pattern: str):
...
+        user: NonBotUser,
         *,
         reason: str = "No reason provided",
     ):
-        """Add a user to the blacklist"""
-        if user.bot:
-            return await ctx.send("You cannot add a bot to the blocklist.")
-        elif await self.bot.is_owner(user):
-            return await ctx.send("You cannot add an owner to the blocklist.")
+        r"""Add a user to the blacklist
```
Comment (Kreusada) [inline advancedblacklist/advanced_blacklist.py:303]:
```suggestion
        """Add a user to the blacklist
```

File: advancedblacklist/converters.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    async def convert(self, ctx: commands.Context, arg: str):
+        maybe_user = await self.converter_type().convert(ctx, arg)
+        if maybe_user.bot:
+            raise commands.BadArgument("That user is a bot")
+        elif maybe_user == ctx.author:
+            raise commands.BadArgument("You cannot blacklist yourself")
+        elif await ctx.bot.is_owner(maybe_user):
+            raise commands.BadArgument("You cannot blacklist this bot's owner")
+        return maybe_user
```
Comment (Kreusada) [inline advancedblacklist/converters.py:22]: Unless I'm being blind, this isn't being used??

**Review by Kreusada — APPROVED** [review]
Looks fine, one small question

File: advancedblacklist/converters.py (hunk 3 above)
Comment (Just-Jojo) [reply advancedblacklist/converters.py:22]: You are correct! I wasn't using that >.< thanks for pointing that out

**Comment by Just-Jojo** [comment]
Thanks for the review!

**Pull request merged**